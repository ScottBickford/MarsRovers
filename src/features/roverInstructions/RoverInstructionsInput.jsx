import React, { useEffect, useRef } from 'react';
import { Segment, Header, Divider, Button } from 'semantic-ui-react';

const RoverInstructionsInput = ({
  instructions,
  setInstructions,
  handleSubmitInstructions
}) => {
  useEffect(() => {
    if (inputRef.current != null) {
      inputRef.current.focus();
    }
  }, []);

  const inputRef = useRef(null);

  return (
    <form onSubmit={handleSubmitInstructions}>
      <Segment.Group>
        <Segment inverted color='teal'>
          <Header>Input</Header>
        </Segment>
        <Segment attached>
          <textarea
            ref={inputRef}
            style={{ width: '100%' }}
            rows='10'
            placeholder='Enter your instructions here...'
            value={instructions}
            onChange={e => setInstructions(e.target.value)}
          />
          <Divider />
          <Button
            content='Submit'
            // disabled={instructions === null || instructions === ''}
          />
        </Segment>
      </Segment.Group>
    </form>
  );
};

export default RoverInstructionsInput;
