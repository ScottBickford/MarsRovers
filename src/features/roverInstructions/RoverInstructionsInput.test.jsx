import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RoverInstructionsInput from './RoverInstructionsInput';

describe('RoverInstructionsInput', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RoverInstructionsInput instructions='5 5\n1 2 N\nLMLMLMLMM\n3 3 E\nMMRMMRMRRM' />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Input', () => {
    const { getByText } = render(<RoverInstructionsInput instructions='5 5\n1 2 N\nLMLMLMLMM\n3 3 E\nMMRMMRMRRM' />);
    const buttonElement = getByText(/Submit/i);
    expect(buttonElement).toBeInTheDocument();
  });
});