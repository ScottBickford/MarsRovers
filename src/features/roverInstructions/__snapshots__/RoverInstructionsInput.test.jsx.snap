// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoverInstructionsInput snapshot renders 1`] = `
<form>
  <div
    className="ui segments"
  >
    <div
      className="ui teal inverted segment"
    >
      <div
        className="ui header"
      >
        Input
      </div>
    </div>
    <div
      className="ui attached segment"
    >
      <textarea
        onChange={[Function]}
        placeholder="Enter your instructions here..."
        rows="10"
        style={
          Object {
            "width": "100%",
          }
        }
        value="5 5\\\\n1 2 N\\\\nLMLMLMLMM\\\\n3 3 E\\\\nMMRMMRMRRM"
      />
      <div
        className="ui divider"
      />
      <button
        className="ui button"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;
